src/main/java/org/tikv/common/log/SlowLogImpl.java

```diff
@@ -38,13 +38,19 @@
   private Throwable error = null;
 
   private final long startMS;
```

Comment:
should it be removed, since we already have `startNano` to replace it?

Reply:
we will lose the time point information if removed.
` System.nanoTime()` can be only used to measure the difference of time.

Reply:
Looks like a little messy when they are used together. And we have to record 2 time (`start ms`, `start ns`) for an event, it increases the latency and use complicity. Is there any best practice to combine these two methods in Java? For example, wrap a TimeRecorder like the following:

```java
public class TimeRecorder {
  private final long startMS;
  private final long startNS;
  private final long endMS;
  private final long endNS;

  public TimeRecorder() {}

  public void start() {
    this.startMS = System.currentTimeMillis();
    this.startNS = System.nanoTime();
  }

  public void end() {
    this.endMS = System.currentTimeMillis();
    this.endNS = System.nanoTime();
  }

  public long getStartMS() {
    return this.startMS;
  }

  public long getEndMS() {
    return this.endMS();
  }

  public long getElapsedTimeMS() {
    return this.endNS() - this.startNS;
  }
}
```

And put the TimeRecord in SlowLogImpl and SlowLogSpanImpl

Reply:
fixed